std/traits.d

```diff
@@ -6515,3 +6515,29 @@ unittest
     @Named("abc") int h;
     static assert(hasUDA!(h, Named));
 }
+
+/**
+Returns: $(D true) iff all types $(D T) are the same.
+*/
+template allSame(T...)
+{
+    static if (T.length <= 1)
+    {
+        enum bool allSame = true;
+    }
+    else
+    {
+        enum bool allSame = is(T[0] == T[1]) && allSame!(T[1..$]);
+    }
+}
+
+unittest
```

Comment:
Add /// before unittest to make it a documentation example. All artifacts must have examples.